ui/src/app/shared/components/application-selector/application-selector.tsx

```diff
@@ -5,6 +5,28 @@ import * as models from '../../models';
 import {appInstanceName, appQualifiedName, ComparisonStatusIcon, HealthStatusIcon, OperationPhaseIcon} from '../../../applications/components/utils';
 
 export const ApplicationSelector = ({apps, formApi}: {apps: models.Application[]; formApi: FormFunctionProps}) => {
+    const reorderedApps: models.Application[] = [];
+    const appOfAppsArray = apps.filter(app => app.isAppOfAppsPattern);
+    if (appOfAppsArray.length > 0) {
```

Comment:
Since `reorderedApps` is initialized to an empty array, and apps are only pushed to the array if at least one of the apps uses App of Apps pattern, this array would be empty if a user doesn't have any applications using App of Apps. This would mean no apps are displayed in the sync panel. Here's an example:
<img width="1188" alt="Screen Shot 2022-08-18 at 08 40 09" src="https://user-images.githubusercontent.com/6250584/185436769-6458edef-79fd-47c0-b11a-903f2a39a684.png">

Reply:
Added else case when there is no App of apps at all, it assigns apps to reorderedApps array.
```
    if (appOfAppsArray.length > 0) {
        ......
    } else {
        reorderedApps = apps;
    }